presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -390,6 +390,56 @@ public void testCast()
                 .setSystemProperty("legacy_timestamp", "false")
                 .build();
         assertQuery(session, "SELECT CAST(date(shipdate) AS timestamp) FROM lineitem");
+
+        // Cast all integer types to short decimal
+        assertQuery("SELECT CAST(linenumber_as_tinyint as DECIMAL(2, 0)) FROM lineitem");
+        assertQuery("SELECT CAST(linenumber_as_smallint as DECIMAL(8, 4)) FROM lineitem");
+        assertQuery("SELECT CAST(CAST(linenumber as INTEGER) as DECIMAL(15, 6)) FROM lineitem");
+        assertQuery("SELECT CAST(nationkey as DECIMAL(18, 6)) FROM nation_partitioned");
+
+        // Cast all integer types to long decimal
+        assertQuery("SELECT CAST(linenumber_as_tinyint as DECIMAL(25, 0)) FROM lineitem");
+        assertQuery("SELECT CAST(linenumber_as_smallint as DECIMAL(19, 4)) FROM lineitem");
+        assertQuery("SELECT CAST(CAST(linenumber as INTEGER) as DECIMAL(20, 6)) FROM lineitem");
+        assertQuery("SELECT CAST(nationkey as DECIMAL(22, 6)) FROM nation_partitioned");
+
+        // Casting DOUBLE to DECIMAL and REAL to DECIMAL is not supported yet.
+
+        // Cast double/float to short decimal
+        // assertQuery("SELECT CAST(discount as DECIMAL(8, 4)) FROM lineitem");
+        // assertQuery("SELECT CAST(discount_as_real as DECIMAL(4, 0)) FROM lineitem");
+
+        // Cast double/float to long decimal
+        // assertQuery("SELECT CAST(discount as DECIMAL(21, 4)) FROM lineitem");
+        // assertQuery("SELECT CAST(discount_as_real as DECIMAL(19, 0)) FROM lineitem");
```

Comment:
Uncomment changes

Reply:
Uncommenting these lines (as well as lines 408-409) results in failures in CI build linux-presto-e2e-tests.

`VeloxUserError:  Cast from DOUBLE to DECIMAL(8,4) is not supported cast((discount) as DECIMAL(8,4))
`

Reply:
Yes, please remove the changes. I forgot this is not implemented yet.